Doc/library/sys.rst

```diff
@@ -283,6 +283,10 @@ always available.
    the real path to its executable, :data:`sys.executable` will be an empty string
    or ``None``.
 
+   .. versionchanged:: 3.7
+      Included executable suffix to path, if one is defined for the
+      platform.
```

Comment:
I'd suggest adding a note which the suffix won't be appended when the path will become an invalid link (e.g. symlink can be made without the executable suffix even in cygwin).

Reply:
Right--I wasn't sure whether or not to include that point.  To me it seems "obvious" that that should be the case, but maybe that's worth clarifying.

As for adding a news entry I agree this change probably merits one, though it still only impacts a small number of platforms so I'll let the core team make that determination...